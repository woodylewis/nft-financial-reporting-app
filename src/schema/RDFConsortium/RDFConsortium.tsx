import React, { FunctionComponent } from 'react';
import PropTypes from 'prop-types';

import { Consortium } from 'schema-dts';
import { JsonLd } from 'react-schemaorg';

interface RDFConsortiumProps {
  id: string;
  members: Array<string>;
}

export const RDFConsortium: FunctionComponent<RDFConsortiumProps> = ({ id, members }) => {
  const context = 'https://schema.org';
  const type = 'Consortium';

  return (
    <JsonLd<Consortium>
      item={{
        '@context': context,
        '@id': id,
        '@type': type,
        member: [
          {
            '@id': members[0],
            '@type': 'Organization',
          },
          {
            '@id': members[1],
            '@type': 'Organization',
          },
        ],
      }}
    />
  );
};

RDFConsortium.propTypes = {
  id: PropTypes.string.isRequired,
  members: PropTypes.array.isRequired,
};
