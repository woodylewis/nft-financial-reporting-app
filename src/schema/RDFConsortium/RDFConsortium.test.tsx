import React from 'react';
// import { shallow, ShallowWrapper, mount } from "enzyme";
import { create, ReactTestRenderer } from 'react-test-renderer';

import { RDFConsortium } from './RDFConsortium';

// let wrapper: ShallowWrapper;
let snapshot: ReactTestRenderer;
// let mountedInstance;

beforeEach(() => {
  const members = ['member1'];
  const network = <RDFConsortium id='DSFNetwork' members={members} />;

  snapshot = create(network);
  // wrapper = shallow(network);
  // mountedInstance = mount(network);
});

describe('<RDFConsortium />', () => {
  test('it matches the snapshot', () => {
    expect(snapshot.toJSON()).toMatchSnapshot();
  });
});
