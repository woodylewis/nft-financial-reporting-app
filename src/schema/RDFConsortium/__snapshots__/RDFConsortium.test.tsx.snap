// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RDFConsortium /> it matches the snapshot 1`] = `
<script
  dangerouslySetInnerHTML={
    {
      "__html": "{"@context":"https://schema.org","@id":"DSFNetwork","@type":"Consortium","member":[{"@id":"member1","@type":"Organization"},{"@type":"Organization"}]}",
    }
  }
  type="application/ld+json"
/>
`;
